### ISSUE TITLE
`InstanceName` of plugins produces unknown field error in Kong 2.8.x

### ISSUE THREAD

**Issue by randmonkey** [Kong/kubernetes-ingress-controller#5241]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Current Behavior

When `InstanceName` is specified in `KongPlugin` or `KongClusterPlugin`, the error happens when sending configuration to Kong 2.8.x 
```
 {"name":"invalid declarative configuration","fields":{"plugins":[{"instance_name":"unknown field"}]},"code":14,"message":"declarative config is invalid: {plugins={{instance_name=\"unknown field\"}}}"}
```
This makes Kong gateway stuck on updating configuration.

### Expected Behavior

Valid configuration of plugins (without `instance_name` field) sent to Kong when Kong version is below 3.2.0. 
Ref: 
https://docs.konghq.com/gateway/2.8.x/admin-api/#plugin-object 
https://docs.konghq.com/gateway/3.1.x/admin-api/#plugin-object
https://docs.konghq.com/gateway/3.2.x/admin-api/#plugin-object
The `instance_name` field is first added in Kong 3.2.x.

### Steps To Reproduce

```markdown
1. Create a KongPlugin foo with field InstanceName
2. Create an Ingress referencing this plugin by adding annotation konghq.com/plugins: "foo"
```

### Kong Ingress Controller version

```shell
2.12.2
```

### Kubernetes version

```shell
1.28.0 kind cluster
```

### Anything else?

This does not affect KIC 3.0.x because KIC 3.x only support Kong 3.4.1 and above.

**Comment by randmonkey** [comment]
Fixed in #5250. TODO: add CHANGELOG.

**Comment by pmalek** [comment]
Would it be possible to conditionally add instance name for Kong Gateway versions older than 3.2? Would it increase the complexity in a way that it would significantly increase the maintenance costs?
